### PR TITLE
3.0 stable

### PR DESCRIPTION
**Pull request by badluck81**
This days i migrated to CI and i found a really ugly bug in router.
I tried to setup $routing['directory'] based on subdomains in index.php like this:

if (($_SERVER['HTTP_HOST']=="www.test.com")  || (($_SERVER['HTTP_HOST']=="test.com")) ){
	$routing['directory'] = 'frontend';
}elseif ( $_SERVER['HTTP_HOST']=="backend.test.com" ){
	$routing['directory'] = 'backend';
}

And here started ugly problems, because in system/core/Router.php the directory is not set first and 
$this->_set_routing(); (in Router.php construct) call _set_default_controller() without calling set_directory() first.
So basically, we cannot setup defalt routes while working with subdomains (with folders works).

The hack i did is in MY_Router.php:
<?php
if (!defined("BASEPATH")) exit("No direct script access allowed");
class MY_Router extends CI_Router {
    function __construct($routing) {
		if (is_array($routing) && isset($routing['directory'])) {
				$this->set_directory($routing['directory']);
		}
		parent::__construct();
    }
}
And default routes worked like a charm.

### CODE REVIEW

**Comment by narfbg** [comment]
I don't understand why this is a pull request (and with changes unrelated to the described issue), but nevertheless, the bug that you're describing is already fixed in version 3.0.1.

See issue #3752.
